deepchem/trans/duplicate.py

```diff
@@ -0,0 +1,171 @@
+import logging
+import numpy as np
+from typing import Tuple
+from deepchem.data import Dataset
+from deepchem.trans.transformers import Transformer
+
+logger = logging.getLogger(__name__)
+
+
+class DuplicateBalancingTransformer(Transformer):
+  """Balance binary or multiclass datasets by duplicating rarer class samples.
+
+  This class balances a dataset by duplicating samples of the rarer class so
+  that the sum of all example weights from all classes is the same. (Up to
```

Comment:
I don't think that's what it actually does.  To tries to balance the *number* of samples in each class, not the *sum of weights* for each class.  Although possibly the latter is what it really should do?

Reply:
Ah good catch! I agree that the latter should be the expected behavior. I've fixed this and added an additional test for weight handling

Reply:
Looks good!